src/test/java/com/epam/edumanagementsystem/admin/impl/AcademicClassServiceImplTest.java

```diff
@@ -0,0 +1,150 @@
+        Set<AcademicCourse> academicCourseSet = new HashSet<>();
+        academicCourseSet.add(academicCourse);
+        academicClass.setAcademicCourseSet(academicCourseSet);
+        Set<Teacher> allTeacher = new HashSet<>();
+        allTeacher.add(teacher);
+        academicClass.setTeacher(allTeacher);
+        academicClass.setClassroomTeacher(teacher);
+    }
+
+    @Test
+    void testCreatePositive() {
+        when(academicClassRepository.save(any(AcademicClass.class))).thenReturn(academicClass);
+        AcademicClass actualAcademicClass = academicClassServiceImpl.create(academicClass);
+        assertEquals(academicClass, actualAcademicClass);
+        verify(academicClassRepository, times(1)).save(actualAcademicClass);
+    }
+    @Test
+    void testCreateNegative() {
+        assertThrows(NullPointerException.class, () -> academicClassServiceImpl.create(null));
+    }
+
+    @Test
+    void testGetByIdPositive() {
+        when(academicClassRepository.findById(1L)).thenReturn(Optional.of(academicClass));
+        AcademicClassDto academicClassDto = academicClassServiceImpl.getById(1L);
+        assertEquals(1L, academicClassDto.getId());
+    }
+
+    @Test
+    void testGetByIdNull() {
+        Long classId = null;
+        when(academicClassRepository.findById(any())).thenReturn(Optional.empty());
+        AcademicClassDto academicClassDto = academicClassServiceImpl.getById(classId);
+        assertEquals(classId, academicClassDto.getId());
+    }
+
+    @Test
+    void testFindByNamePositive() {
+        when(academicClassRepository.findByClassNumber("A1")).thenReturn(academicClass);
+        AcademicClass actualClass = academicClassServiceImpl.findByName(academicClass.getClassNumber());
+        assertEquals(academicClass.getClassNumber(), actualClass.getClassNumber());
+    }
+
+    @Test
+    void testFindByNameNegative() {
+        when(academicClassRepository.findByClassNumber("A1")).thenReturn(academicClass);
+        AcademicClass aClass = new AcademicClass(1L, "A3", null, null, null, null, null);
+        AcademicClass actualClass = academicClassServiceImpl.findByName(academicClass.getClassNumber());
+        assertNotEquals(actualClass.getClassNumber(), aClass.getClassNumber());
+    }
+
+    @Test
+    void testUpdatePositive() {
+        Mockito.when(academicClassRepository.findByClassNumber("A1")).thenReturn(academicClass);
+        Mockito.when(academicClassRepository.save(academicClass)).thenReturn(academicClass);
+        AcademicClass academicClass1 = academicClassServiceImpl.update(academicClass);
+        academicClassServiceImpl.update(academicClass1);
+        assertNotNull(academicClass);
+        assertNotNull(academicClass1);
+        assertEquals(academicClass.getClassNumber(), academicClass1.getClassNumber());
+    }
+
+    @Test
+    void testFindAllAcademicCourses() {
+        Mockito.when(academicClassRepository.findByClassNumber("A1")).thenReturn(academicClass);
```

Comment:
Whenever the method under the test returns a list or any other collection, you also should return a collection in thenReturn part, like thenReturn(List.of(academicClass). And after it check if the actual list returned by the method is not null and the size is equal to the expected list's size or whatever you wanted to test.

Reply:
thenReturn(List.of(academicClass) It's not going to work, here I can use thenReturn(new AcademicClass()), but after that I will get NullPointerException.

Reply:
In that case get academic courses from found academic class map it to list and use it in asserts